src/clj/game/cards/events.clj

```diff
@@ -50,22 +64,23 @@
                        (gain state :runner :memory (:memoryunits c))))))}
 
    "Blackmail"
-   {:req (req has-bad-pub) :prompt "Choose a server" :choices (req runnable-servers)
-    :msg "prevent ICE from being rezzed during this run"
-    :effect (effect (register-run-flag!
-                      card
-                      :can-rez
-                      (fn [state side card]
-                        (if (ice? card)
-                          ((constantly false) (system-msg state side (str "is prevented from rezzing ICE on this run by Blackmail")))
-                          true)))
-                    (run target nil card))}
+   (run-event
+    {:req (req has-bad-pub)
+     :msg "prevent ICE from being rezzed during this run"}
+    nil
+    (effect (register-run-flag!
+             card
+             :can-rez
+             (fn [state side card]
+               (if (ice? card)
+                 ((constantly false) (system-msg state side (str "is prevented from rezzing ICE on this run by Blackmail")))
```

Comment:
This was the first run-flag card, and still writes to the game log instead of using a toast like everything else now does.